src/nvim/eval.c

```diff
@@ -6962,9 +6963,12 @@ void set_buffer_lines(buf_T *buf, linenr_T lnum_arg, bool append,
 
     if (!append && lnum <= curbuf->b_ml.ml_line_count) {
       // Existing line, replace it.
+      int old_len = (int)STRLEN(ml_get(lnum));
       if (u_savesub(lnum) == OK
           && ml_replace(lnum, (char_u *)line, true) == OK) {
         changed_bytes(lnum, 0);
```

Comment:
you could use `inserted_bytes` I think (and replace both this and next line)

Reply:
The thing that obthers me is that this overlaps with `changed_bytes` below

Reply:
Should be all good now